test/classification/training/test_trainer.py

```diff
@@ -216,6 +216,35 @@ def test_warmup(self):
         trainer.fit(model, [dataloaders[0]])
         self.assertEqual(trainer.warmup_steps, 1)
 
+    def test_save_load(self):
+        fd, checkpoint_path = tempfile.mkstemp()
```

Comment:
can we put this in a `try`/`except`/`finally` or use a context `with tempfile.NamedTemporaryFile() as f:` to ensure proper cleanup?

Reply:
I put a NamedTemporaryFile, however, the tempfile.mkstemp() I copied from test_save_load() from test_multitask_classifier.py. Maybe you want to update it there, too?

Reply:
got it — yes, we likely need to clean up other parts of the codebase as well. :)